////////////////////////////////////////////////////////////////////////////////////////////////////
///
/// @file      AudioSettings.proto
///
/// @attention Copyright (C) 2017 Bose Corporation All Rights Reserved
///
///            Bose Corporation
///            The Mountain Road,
///            Framingham, MA 01701-9168
///            U.S.A.
///
///            This program may not be reproduced, in whole or in part, in any form by any means
///            whatsoever without the written permission of Bose Corporation.
///
////////////////////////////////////////////////////////////////////////////////////////////////////
package ProductPb;
import "ContentItem.proto";

message AudioProperties
{
    optional int32 min = 1;
    optional int32 max = 2;
    optional uint32 step = 3;
    repeated string supportedPersistence = 4;
}

message AudioBassLevel
{
    optional int32 value = 1;
    optional string persistence = 2;
    optional ContentItem contentItem = 3;
    optional AudioProperties properties = 4;
}
message AudioTrebleLevel
{
    optional int32 value = 1;
    optional string persistence = 2;
    optional ContentItem contentItem = 3;
    optional AudioProperties properties = 4;
}
message AudioCenterLevel
{
    optional int32 value = 1;
    optional string persistence = 2;
    optional ContentItem contentItem = 3;
    optional AudioProperties properties = 4;
}
message AudioSurroundLevel
{
    optional int32 value = 1;
    optional string persistence = 2;
    optional ContentItem contentItem = 3;
    optional AudioProperties properties = 4;
}
message AudioGainOffset
{
    optional int32 value = 1;
    optional string persistence = 2;
    optional ContentItem contentItem = 3;
    optional AudioProperties properties = 4;
}
message AudioAvSync
{
    optional int32 value = 1;
    optional string persistence = 2;
    optional ContentItem contentItem = 3;
    optional AudioProperties properties = 4;
}
message AudioMode
{
    message AudioModeProperties
    {
        repeated string supportedValues = 1;
        repeated string supportedPersistence = 2;
    }
    optional string value = 1;
    optional string persistence = 2;
    optional ContentItem contentItem = 3;
    optional AudioModeProperties properties = 4;
}
message AudioContentType
{
    message ContentTypeProperties
    {
        repeated string supportedValues = 1;
        repeated string supportedPersistence = 2;
    }
    optional string value = 1;
    optional string persistence = 2;
    optional ContentItem contentItem = 3;
    optional ContentTypeProperties properties = 4;
}
